bunch-cli/src/main/kotlin/org/jetbrains/bunches/git/ignore.kt

```diff
@@ -24,5 +45,5 @@ fun isGitRoot(dir: File) = File(dir, ".git").exists()
 fun isNestedGitRoot(dir: File, baseGitRoot: File) =
     isGitRoot(dir) && dir != baseGitRoot
 
-fun shouldIgnoreDir(dir: File, baseGitRoot: File) =
-    isGitDir(dir) || isOutDir(dir) || isGradleBuildDir(dir) || isGradleDir(dir) || isNestedGitRoot(dir, baseGitRoot)
+fun shouldIgnoreDir(dir: File, baseGitRoot: File, ignore: IgnoreNode?) =
+    isGitDir(dir) || isOutDir(dir) || isGradleBuildDir(dir) || isGradleDir(dir) || checkIgnoreList(dir, ignore) || isNestedGitRoot(dir, baseGitRoot)
```

Comment:
I would try to replace all other checks with git ignore. Do you know what performance characteristics for `checkIgnoreList` are?

Reply:
In this function we only check all rules from `ignore` using RegExp matcher (parsing of file isn't included in this function), so it should work in linear time of amount of ingore rules in `.gitignore` multiplied by the complexity of RegExp matcher.